go-controller/pkg/ovn/pods.go

```diff
@@ -563,23 +560,30 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	if err != nil {
 		return fmt.Errorf("error creating logical switch port %+v on switch %+v: %+v", *lsp, *ls, err)
 	}
+	allOps = append(allOps, ops...)
```

Comment:
@tssurya CreateOrUpdateLogicalSwitchPortsOnSwitchOps() takes the `ops` parameter and appends those to the ops it returns, so doing the append to allops here will generate duplicate operations in our transaction. That's why some of the CI tests failed to set up kind:

```
E0523 10:51:30.213494      66 obj_retry.go:1464] Failed to update resource *v1.Pod, old=local-path-storage/local-path-provisioner-5ddd94ff66-57vxd, new=local-path-storage/local-path-provisioner-5ddd94ff66-57vxd, error: addLogicalPort failed for local-path-storage/local-path-provisioner-5ddd94ff66-57vxd: error transacting operations [{Op:mutate Table:Address_Set Row:map[] Rows:[] Columns:[] Mutations:[{Column:addresses Mutator:insert Value:{GoSet:[10.244.2.5]}}] Timeout:<nil> Where:[where column _uuid == {bbba2901-777e-435c-ba10-f616792ce356}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:NAT Row:map[external_ip:172.18.0.3 logical_ip:10.244.2.5 options:{GoMap:map[stateless:false]} type:snat] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996282} {Op:mutate Table:Logical_Router Row:map[] Rows:[] Columns:[] Mutations:[{Column:nat Mutator:insert Value:{GoSet:[{GoUUID:u2596996282}]}}] Timeout:<nil> Where:[where column _uuid == {60899c13-9f06-4c26-bbf8-91f3c8c1ba39}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:NAT Row:map[external_ip:172.18.0.3 logical_ip:10.244.2.5 options:{GoMap:map[stateless:false]} type:snat] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996282} {Op:mutate Table:Logical_Router Row:map[] Rows:[] Columns:[] Mutations:[{Column:nat Mutator:insert Value:{GoSet:[{GoUUID:u2596996282}]}}] Timeout:<nil> Where:[where column _uuid == {60899c13-9f06-4c26-bbf8-91f3c8c1ba39}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:Logical_Switch_Port Row:map[addresses:{GoSet:[0a:58:0a:f4:02:05 10.244.2.5]} external_ids:{GoMap:map[namespace:local-path-storage pod:true]} name:local-path-storage_local-path-provisioner-5ddd94ff66-57vxd options:{GoMap:map[iface-id-ver:8831a376-2e23-4790-9bca-a32b50185454 requested-chassis:ovn-control-plane]} port_security:{GoSet:[0a:58:0a:f4:02:05 10.244.2.5]}] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996283} {Op:mutate Table:Logical_Switch Row:map[] Rows:[] Columns:[] Mutations:[{Column:ports Mutator:insert Value:{GoSet:[{GoUUID:u2596996283}]}}] Timeout:<nil> Where:[where column _uuid == {6821a94b-f633-49f8-be5b-c20f8caf24dd}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:}]: error in transact with ops [{Op:mutate Table:Address_Set Row:map[] Rows:[] Columns:[] Mutations:[{Column:addresses Mutator:insert Value:{GoSet:[10.244.2.5]}}] Timeout:<nil> Where:[where column _uuid == {bbba2901-777e-435c-ba10-f616792ce356}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:NAT Row:map[external_ip:172.18.0.3 logical_ip:10.244.2.5 options:{GoMap:map[stateless:false]} type:snat] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996282} {Op:mutate Table:Logical_Router Row:map[] Rows:[] Columns:[] Mutations:[{Column:nat Mutator:insert Value:{GoSet:[{GoUUID:u2596996282}]}}] Timeout:<nil> Where:[where column _uuid == {60899c13-9f06-4c26-bbf8-91f3c8c1ba39}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:NAT Row:map[external_ip:172.18.0.3 logical_ip:10.244.2.5 options:{GoMap:map[stateless:false]} type:snat] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996282} {Op:mutate Table:Logical_Router Row:map[] Rows:[] Columns:[] Mutations:[{Column:nat Mutator:insert Value:{GoSet:[{GoUUID:u2596996282}]}}] Timeout:<nil> Where:[where column _uuid == {60899c13-9f06-4c26-bbf8-91f3c8c1ba39}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:insert Table:Logical_Switch_Port Row:map[addresses:{GoSet:[0a:58:0a:f4:02:05 10.244.2.5]} external_ids:{GoMap:map[namespace:local-path-storage pod:true]} name:local-path-storage_local-path-provisioner-5ddd94ff66-57vxd options:{GoMap:map[iface-id-ver:8831a376-2e23-4790-9bca-a32b50185454 requested-chassis:ovn-control-plane]} port_security:{GoSet:[0a:58:0a:f4:02:05 10.244.2.5]}] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996283} {Op:mutate Table:Logical_Switch Row:map[] Rows:[] Columns:[] Mutations:[{Column:ports Mutator:insert Value:{GoSet:[{GoUUID:u2596996283}]}}] Timeout:<nil> Where:[where column _uuid == {6821a94b-f633-49f8-be5b-c20f8caf24dd}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:}] results [{Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error: Details: UUID:{GoUUID:41cd19f8-8878-4dc7-a723-936671ed1929} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error:duplicate uuid-name Details:This "uuid-name" appeared on an earlier "insert" operation. UUID:{GoUUID:} Rows:[]} {Count:0 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error: Details: UUID:{GoUUID:} Rows:[]}] and errors [duplicate uuid-name: This "uuid-name" appeared on an earlier "insert" operation.]: 1 ovsdb operations failed
```

I think let's just kill `allOps` and do the function-appends-to-and-returns-ops thing that libovsdbops wants to do now.

Reply:
oh this is why ops was getting overwritten and I was wondering why?! that makes sense :)